.github/workflows/codeql.yml

```diff
@@ -57,20 +57,11 @@ jobs:
           # By default, queries listed here will override any specified in a config file.
           # Prefix the list here with "+" to use these queries and those in the config file.
 
-      # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
-      # If this step fails, then you should remove it and run the build manually (see below)
-      - name: Autobuild
-        uses: github/codeql-action/autobuild@b374143c1149a9115d881581d29b8390bbcbb59c # v3.22.11
-
-      # ℹ️ Command-line programs to run using the OS shell.
-      # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
-
-      #   If the Autobuild fails above, remove it and uncomment the following three lines.
-      #   modify them (or add more) to build your code if your project, please refer to the EXAMPLE below for guidance.
-
-      # - run: |
-      #   echo "Run, Build Application using script"
-      #   ./location_of_script_within_repo/buildscript.sh
+      # NOTE: Removed autobuild step as it was leading to hanging in Github Actions
```

Comment:
I have removed the `autobuild` step from here (as the comments instructed) as it is leading to behaviour that left the step indefinitely hanging (see [here](https://github.com/in-toto/witness/actions/runs/7251758704/job/19754767599)). In my opinion such exhaustive use of compute time should be avoided at all costs.

I think it would be nice to get back to using the `autobuild` scripting, however it does introduce some guess work and ambiguity that can be eradicated by simply pointing the workflow at `make` or `make build`. I also don't believe that this change sacrifices any functionality.

@jkjell @kairoaraujo would be good to get opinions here. I also recognise that this PR may not be the place for such a change, if you want I can go ahead and move it to a separate PR.

Reply:
I'm not sure if this was from the job or because we're an OSS project but, the billable time was `0 s` for the failure: https://github.com/in-toto/witness/actions/runs/7251758704/usage. It looks like the only thing the autobuild did was call make. So, that **makes** (😂) me think we could hit the same issue with a direct make command too.

Reply:
But, yeah, if we want this change ➕ to a separate PR.

Reply:
Yeah I see what you mean by us being an OSS project, still gives me the heebiejeebies.

As for what caused it I am still not totally sure, but the `autobuild` logic actually performs `make` (which fails obviously) and continues anyway: 

```
 2023/12/18 17:47:27 Running /usr/bin/make failed, continuing anyway: exit status 2
  2023/12/18 17:47:27 Build failed, continuing to install dependencies.
  2023/12/18 17:47:27 Installing dependencies using `go get -v ./...` in `.`.
 ```
 
 Whatever the matter is, I feel it's more deterministic to do it this way, so I reckon we should split this out ot a separate PR regardless. I don't think we should trust any automation that we don't necessarily need? If it wants to know how we build, we can just tell it 😄 .